rust/rope/src/diff.rs

```diff
@@ -65,6 +66,18 @@ impl Diff<RopeInfo> for LineHashDiff {
             return builder.to_delta(base, target);
         }
 
+        // if a continuous range of text got deleted, we're done
+        if target.len() < base.len() && start_offset + diff_end == target.len() {
+            builder.copy(base.len() - diff_end, target_end, diff_end);
+            return builder.to_delta(base, target);
+        }
+
+        // if a continuous range of text got inserted, we're done
+        if target.len() > base.len() && start_offset + diff_end == base.len() {
+            builder.copy(base.len() - diff_end, target_end, diff_end);
+            return builder.to_delta(base, target);
+        }
```

Comment:
It might make sense to have this be a separate patch, since it doesn't seem to be really related to the other changes here, and it might be worth having anyway? It would be nice to measure and see that this is a reasonable speedup in these cases (it makes sense that it would be, but it's nice to measure this sort of code).

Reply:
I've separated this in #1295. It _is_ an improvement.